tests/integration/sharding_tests/test_sharding_relations.py

```diff
@@ -0,0 +1,139 @@
+            SHARD_ONE_APP_NAME,
+        ],
+        idle_period=20,
+        raise_on_blocked=False,
+        timeout=TIMEOUT,
+    )
+
+
+async def test_only_one_config_server_relation(ops_test: OpsTest) -> None:
+    """Verify that a shard can only be related to one config server."""
+    await ops_test.model.integrate(
+        f"{SHARD_ONE_APP_NAME}:{SHARD_REL_NAME}",
+        f"{CONFIG_SERVER_ONE_APP_NAME}:{CONFIG_SERVER_REL_NAME}",
+    )
+
+    multiple_config_server_rels_allowed = True
+    try:
+        await ops_test.model.integrate(
+            f"{SHARD_ONE_APP_NAME}:{SHARD_REL_NAME}",
+            f"{CONFIG_SERVER_TWO_APP_NAME}:{CONFIG_SERVER_REL_NAME}",
+        )
+    except JujuAPIError as e:
+        if e.error_code == "quota limit exceeded":
+            multiple_config_server_rels_allowed = False
+
+    assert not multiple_config_server_rels_allowed, "Shard can relate to multiple config servers."
+
+
+async def test_cannot_use_db_relation(ops_test: OpsTest) -> None:
+    """Verify that a shard cannot use the DB relation."""
+    await ops_test.model.integrate(
+        f"{APP_CHARM_NAME}:{DATABASE_REL_NAME}",
+        SHARD_ONE_APP_NAME,
+    )
+
+    await ops_test.model.wait_for_idle(
+        apps=[SHARD_ONE_APP_NAME],
+        idle_period=20,
+        raise_on_blocked=False,
+        timeout=TIMEOUT,
+    )
+
+    shard_unit = ops_test.model.applications[SHARD_ONE_APP_NAME].units[0]
+    assert (
+        shard_unit.workload_status_message == "Sharding roles do not support database interface."
+    ), "Shard cannot be related using the database relation"
+
+    await ops_test.model.applications[SHARD_ONE_APP_NAME].remove_relation(
+        f"{APP_CHARM_NAME}:{DATABASE_REL_NAME}",
+        SHARD_ONE_APP_NAME,
+    )
+
+    await ops_test.model.wait_for_idle(
+        apps=[SHARD_ONE_APP_NAME],
+        idle_period=20,
+        raise_on_blocked=False,
+        timeout=TIMEOUT,
+    )
+
+
+async def test_cannot_use_legacy_db_relation(ops_test: OpsTest) -> None:
+    """Verify that a shard cannot use the DB relation."""
+    await ops_test.model.integrate(
+        LEGACY_APP_CHARM_NAME,
+        SHARD_ONE_APP_NAME,
+    )
+
+    await ops_test.model.wait_for_idle(
+        apps=[SHARD_ONE_APP_NAME],
+        idle_period=20,
+        raise_on_blocked=False,
+        timeout=TIMEOUT,
+    )
+
+    shard_unit = ops_test.model.applications[SHARD_ONE_APP_NAME].units[0]
+    assert (
+        shard_unit.workload_status_message == "Sharding roles do not support obsolete interface."
+    ), "Shard cannot be related using the mongodb relation"
+
+    await ops_test.model.applications[SHARD_ONE_APP_NAME].remove_relation(
+        f"{SHARD_ONE_APP_NAME}:{LEGACY_RELATION_NAME}",
+        f"{LEGACY_APP_CHARM_NAME}:{LEGACY_RELATION_NAME}",
+    )
+
+    await ops_test.model.wait_for_idle(
+        apps=[SHARD_ONE_APP_NAME],
+        idle_period=20,
+        raise_on_blocked=False,
+        timeout=TIMEOUT,
+    )
```

Comment:
If I get it right, the above two commands are only to clean up.
Why not to indicate that with a comment?
(Just to make it clear that we didn't expect any specific outcome to verify after setting up this state.)